arlas-core/src/main/java/io/arlas/server/core/FluidSearch.java

```diff
@@ -729,6 +731,34 @@ private ValuesSourceAggregationBuilder setAggeragatedGeometryStrategy(Aggregatio
         return aggregationBuilder;
     }
 
+    private ValuesSourceAggregationBuilder setHitsToFetch(Aggregation aggregationModel, ValuesSourceAggregationBuilder aggregationBuilder) throws ArlasException {
+        if (aggregationModel.fetchHits != null) {
+            TopHitsAggregationBuilder topHitsAggregationBuilder = AggregationBuilders.topHits("fetched_hits");
+            Integer size = aggregationModel.fetchHits.size != null ? aggregationModel.fetchHits.size : 1;
+            topHitsAggregationBuilder.size(size);
+
+            List<String> includes = new ArrayList<>();
+            if (aggregationModel.fetchHits.include != null) {
+                aggregationModel.fetchHits.include.forEach(field ->{
+                    if (field.startsWith("+")) {
+                        includes.add(field.substring(1));
+                        topHitsAggregationBuilder.sort(field.substring(1), SortOrder.ASC);
+                    } else if (field.startsWith("-")) {
+                        includes.add(field.substring(1));
+                        topHitsAggregationBuilder.sort(field.substring(1), SortOrder.DESC);
+                    } else {
+                        includes.add(field);
+                    }
+                });
+                String[] hitsToInclude = new String[includes.size()];
+                hitsToInclude = includes.toArray(hitsToInclude);
```

Comment:
You can probably instanciate the array in one line, like includes.toArray(new String[includes.size()])